src/Hosting/Server.IntegrationTesting/src/Deployers/SelfHostDeployer.cs

```diff
@@ -55,23 +56,34 @@ public override async Task<DeploymentResult> DeployAsync()
                     DotnetPublish();
                 }
 
-                var hintUrl = TestUriHelper.BuildTestUri(
-                    DeploymentParameters.ServerType,
-                    DeploymentParameters.Scheme,
-                    DeploymentParameters.ApplicationBaseUriHint,
-                    DeploymentParameters.StatusMessagesEnabled);
-
                 // Launch the host process.
-                var (actualUrl, hostExitToken) = await StartSelfHostAsync(hintUrl);
+                for (var i = 0; i < RetryCount; i++)
+                {
+                    var hintUrl = TestUriHelper.BuildTestUri(
+                        DeploymentParameters.ServerType,
+                        DeploymentParameters.Scheme,
+                        DeploymentParameters.ApplicationBaseUriHint,
+                        DeploymentParameters.StatusMessagesEnabled);
+                    var (actualUrl, hostExitToken) = await StartSelfHostAsync(hintUrl);
+
+                    if (DeploymentParameters.ServerType == ServerType.HttpSys)
```

Comment:
This doesn't actually check if the app successfully started or not before retrying.